python/ray/_private/gcs_utils.py

```diff
@@ -197,12 +205,16 @@ def wrapper(self, *args, **kwargs):
                 try:
                     return f(self, *args, **kwargs)
                 except grpc.RpcError as e:
-                    if remaining_retry <= 0:
-                        raise
                     if e.code() in (
                         grpc.StatusCode.UNAVAILABLE,
                         grpc.StatusCode.UNKNOWN,
                     ):
+                        if remaining_retry <= 0:
+                            logger.error(
+                                "Failed to connect to GCS. Please check"
+                                " `gcs_server.out` for more details."
+                            )
+                            raise
```

Comment:
Keep the RpcError since other code depends on this exception. Using logger to print more infos.